scripts/src/cowidev/yougov/__main__.py

```diff
@@ -271,17 +271,19 @@ def _standardize_entities(df):
 
 def _aggregate(df):
     s_period = df["date"].dt.to_period(FREQ)
-    df.loc[:, "date_end"] = s_period.dt.end_time.dt.date
+    df.loc[:, "date_mid"] = (
+        s_period.dt.start_time + (s_period.dt.end_time - s_period.dt.start_time) / 2
+    ).dt.date
```

Comment:
Suggestion: use the 15th day as the middle of the month for all months, which may conform better with popular intuitions of a month's midpoint (compared to January 16th, February 14th, etc as assigned by retrieving the actual midpoint). Using the 15th arguably looks better in charts too (for consistency).

```suggestion
if FREQ == 'M':
    df.loc[:, "date_mid"] = s_period.dt.start_time.dt.date + datetime.timedelta(days=14)
else:
    df.loc[:, "date_mid"] = (
        s_period.dt.start_time + (s_period.dt.end_time - s_period.dt.start_time) / 2
    ).dt.date
```